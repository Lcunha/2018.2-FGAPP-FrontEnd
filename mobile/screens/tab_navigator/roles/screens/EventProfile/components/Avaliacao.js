import React, { Component } from "react"
import {
	Card,
	CardItem,
	Left,
	Body,
	Right,
	List,
	ListItem,
	Icon,
	Button
} from "native-base"
import {
	View,
	ScrollView,
	Text,
	StyleSheet,
	Linking,
	TouchableOpacity
} from "react-native"

import Divider from './Divider'
import CardHeader from './CardHeader'

class Avaliacao extends Component {
	state = {
		thumbsUp: false,
		thumbsDown: false
	}

	_toggleThumbs(thumbs) {
		if (thumbs == "up") {
			if (this.state.thumbsDown)
				this.setState({ thumbsDown: !this.state.thumbsDown })
			this.setState({ thumbsUp: !this.state.thumbsUp })
		} else if (thumbs == "down") {
			if (this.state.thumbsUp)
				this.setState({ thumbsUp: !this.state.thumbsUp })
			this.setState({ thumbsDown: !this.state.thumbsDown })
		}
	}

	render() {
		return (
			<Card>
				<CardHeader text="Avaliação"/>
				<Body>
					<Card transparent style={styles.subCards}>
						<Text style={styles.subHeaders}>
							Como está o evento?
						</Text>

						<CardItem style={{ alignSelf: "center", marginBottom:-30}}>
							<Left>
								<TouchableOpacity
									onPress={() => this._toggleThumbs("up")}
								>
									<Icon
										name="thumbs-up"
										style={{
											fontSize: 40,
											color: this.state.thumbsUp
												? "green"
												: "black"
										}}
									/>
								</TouchableOpacity>

								<Text style={styles.likeDislikeBox}>11</Text>
							</Left>

							<Right>
								<CardItem>
									<Text style={styles.likeDislikeBox}>4</Text>
									<TouchableOpacity
										onPress={() =>
											this._toggleThumbs("down")
										}
									>
										<Icon
											name="thumbs-down"
											style={{
												fontSize: 40,
												color: this.state.thumbsDown
													? "red"
													: "black"
											}}
										/>
									</TouchableOpacity>
								</CardItem>
							</Right>
						</CardItem>
					</Card>
				</Body>
			</Card>
		)
	}
}

const styles = StyleSheet.create({
	likeDislikeBox: {
		borderWidth: 1,
		borderRadius: 5,
		width: 50,
		height: 30,

		alignSelf: "center",
		justifyContent: "center",
		textAlign: "center",
		margin: 20,
		padding: 5
	},
	subCards: {
		alignSelf: "center",
		width: 300
	},
	subHeaders: {
		alignSelf: "center",
		fontSize: 15
	},
	btn: {
		margin: 7
	},
})

export default Avaliacao
