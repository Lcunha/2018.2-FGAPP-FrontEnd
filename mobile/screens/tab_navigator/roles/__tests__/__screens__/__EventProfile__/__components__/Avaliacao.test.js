import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Avaliacao from '../../../../screens/EventProfile/components/Avaliacao';

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer.create(<Avaliacao />).toJSON();
  expect(tree).toMatchSnapshot();
});
