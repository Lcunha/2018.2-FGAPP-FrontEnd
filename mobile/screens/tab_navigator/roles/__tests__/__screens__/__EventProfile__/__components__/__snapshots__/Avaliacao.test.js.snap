// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderLeftWidth": 0.5,
      "borderRadius": 2,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexWrap": "nowrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
  virtual={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "grey",
        "marginBottom": 10,
      }
    }
  >
    Avaliação
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomWidth": 0,
            "borderColor": "#ccc",
            "borderLeftWidth": 0,
            "borderRadius": 2,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "elevation": null,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          },
          Object {
            "alignSelf": "center",
            "width": 300,
          },
        ]
      }
      transparent={true}
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 15,
          }
        }
      >
        Como está o evento?
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            },
            Object {
              "alignSelf": "center",
              "marginBottom": -30,
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
          virtual={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 30,
                "justifyContent": "center",
                "margin": 20,
                "padding": 5,
                "textAlign": "center",
                "width": 50,
              }
            }
          >
            11
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 0.8,
            }
          }
          virtual={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 30,
                  "justifyContent": "center",
                  "margin": 20,
                  "padding": 5,
                  "textAlign": "center",
                  "width": 50,
                }
              }
            >
              4
            </Text>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
